common/remote/sockfile.cpp

```diff
@@ -2719,42 +2732,78 @@ ISocket *connectDafs(SocketEndpoint &ep, unsigned timeoutms)
     if ( (securitySettings.connectMethod == SSLNone) || (securitySettings.connectMethod == SSLOnly) )
     {
         socket.setown(ISocket::connect_timeout(ep, timeoutms));
-        return socket.getClear();
+        return checkSocketSecure(socket);
     }
 
+    // SSLFirst or UnsecureFirst ...
+
     unsigned newtimeout = timeoutms;
     if (newtimeout > 5000)
         newtimeout = 5000;
 
-    bool tryAgain = false;
-    try
-    {
-        socket.setown(ISocket::connect_timeout(ep, newtimeout));
-    }
-    catch (IJSOCK_Exception *e)
+    int conAttempts = 2;
+    while (conAttempts > 0)
     {
-        if (e->errorCode() == JSOCKERR_connection_failed)
+        conAttempts--;
+        bool connected = false;
+        try
+        {
+            socket.setown(ISocket::connect_timeout(ep, newtimeout));
```

Comment:
I debated this a while and there are several trade-offs.  In the extra port connection attempt here (when configured for SSLFirst or UnsecureFirst) we use a 5 second timeout and if that fails we revert to the original timeout passed in, so this could exceed the original timeout by a maximum of 5 seconds.  Would using up to half the original connection time passed in be a better solution ?  I am just not sure.